hedera-node/hedera-mono-service/src/main/java/com/hedera/services/config/AccountNumbers.java

```diff
@@ -52,76 +54,87 @@ public AccountNumbers(@CompositeProps PropertySource properties) {
         this.properties = properties;
     }
 
+    @Override
     public long treasury() {
         if (treasury == UNKNOWN_NUMBER) {
```

Comment:
is it ok that the class is not thread safe?

Reply:
It will be a singleton class and loaded once at start up.